examples/energy-management-app/energy-management-common/include/ChargingTargetsMemMgr.h

```diff
@@ -0,0 +1,160 @@
+namespace app {
+namespace Clusters {
+namespace EnergyEvse {
+
+/*
+ * The full Target data structure defined as:
+ *
+ * DataModel::List<const Structs::ChargingTargetScheduleStruct::Type> chargingTargetSchedules;
+ *
+ * contains a list of ChargingTargetScheduleStructs which in turn contains a list of ChargingTargetStructs.
+ * This means that somewhere the following memory needs to be allocated in the case where
+ * the Target is at its maximum size:
+ *
+ * ChargingTargetStruct::Type mDailyChargingTargets[kEvseTargetsMaxNumberOfDays][kEvseTargetsMaxTargetsPerDay]
+ *
+ * This is 1680B.
+ *
+ * However it is likely the number of chargingTargets configured will be considerably less. To avoid
+ * allocating the maximum possible Target size, each List<ChargingTargetScheduleStructs> is allocated
+ * separately. This class handles that allocation.
+ *
+ * When iterating through the chargingTargetSchedules, an index in this list is kept and the
+ * ChargingTargetsMemMgr::Reset must be called so this object knows which day schedule it is tracking.
+ * This will free any previous memory allocated for the day schedule in this object.
+ *
+ * There are then three usage cases:
+ *
+ * 1. When loading the Target from persistent storage. In this scenario, it is not known upfront
+ *    how many chargingTargets are associated with this day schedule so ChargingTargetsMemMgr::AddChargingTarget()
+ *    needs to be called as each individual chargingTarget is loaded from persistent data.
+ *
+ *    Once the chargingTargets for the day schedule have been loaded, ChargingTargetsMemMgr::AllocAndCopy() is
+ *    called to allocate the memory to store the chargingTargets and the chargingTargets are copied.
+ *
+ * 2. When updating a Target and a day schedule is unaffected, the chargingTargets associated with
+ *    day schedule need copying. The following should be called:
+ *
+ *    ChargingTargetsMemMgr::AllocAndCopy(const DataModel::List<const Structs::ChargingTargetStruct::Type> & chargingTargets)
+ *
+ * 3. When in SetTargets, a new list of chargingTargets needs to be added to a day schedule, the following
+ *    should be called:
+ *
+ *    ChargingTargetsMemMgr::AllocAndCopy(const DataModel::DecodableList<Structs::ChargingTargetStruct::DecodableType> &
+ * chargingTargets)
+ *
+ * Having allocated and copied the chargingTargets accordingly, they can be added to a
+ * DataModel::List<const Structs::ChargingTargetStruct::Type as follows:
+ *
+ * chargingTargetsList = DataModel::List<Structs::ChargingTargetStruct::Type>(ChargingTargetsMemMgr::GetChargingTargets(),
+ * ChargingTargetsMemMgr::GetNumDailyChargingTargets());
+ *
+ * All memory allocated by this object is released When the ChargingTargetsMemMgr destructor is called.
+ *
+ */
+
+class ChargingTargetsMemMgr
+{
+public:
+    ChargingTargetsMemMgr();
+    ~ChargingTargetsMemMgr();
+
+    /**
+     * @brief This method sets the index (mChargingTargetSchedulesIdx) to use into mpListOfDays.
```

Comment:
all these comments below reference `mpListOfDays`. That is not a public field and is an implementation detail. 

The documentation should describe what the class does, not the implementation. Unure if maybe just saying `list of days` makes sense, however you should never reference private bits in comments  for public API.